src/Client.php

```diff
@@ -264,23 +264,30 @@ public function getIdealIssuers(): IdealIssuers
      * Only completed orders could be refunded.
      *
      * @param string $order_id
+     * @param Properties\Amount $anount
      * @return array
      * @throws \GingerPluginSdk\Exceptions\InvalidOrderStatusException
      * @throws \GingerPluginSdk\Exceptions\RefundFailedException
      */
-    public function refundOrder(string $order_id)
+    public function refundOrder(string $order_id, Properties\Amount $amount = null)
     {
         $order = $this->getOrder(id: $order_id);
 
         if ($order->getStatus()->get() != "completed") {
             throw new InvalidOrderStatusException($order->getStatus()->get(), 'completed');
         }
 
-        if (!$order->getCurrentTransaction()->isCaptured()) {
+        if ($order->getCurrentTransaction()->isCapturable() && !$order->getCurrentTransaction()->isCaptured()) {
             throw  new RefundFailedException('Order is not yet captured, only captured order could be refunded');
         }
 
-        return $this->api_client->refundOrder(id: $order_id, orderData: ['order_lines' => $order->getOrderLines()->toArray()]);
+        return $this->api_client->refundOrder(
+            id: $order_id,
+            orderData: [
+                'amount' => $amount->get() ?? $order->getAmount()->get(),
+                "description" => "Order refund",
+                'order_lines' => $order->getOrderLines()->toArray()
+            ]);
     }
```

Comment:
I also tested partial refund by drupal plugin. It's work

Reply:
👍🏻